GameServerLib/GameObjects/AttackableUnits/AI/ObjAIBase.cs

```diff
@@ -546,6 +546,14 @@ public override void Update(float diff)
 
             var onUpdate = _scriptEngine.GetStaticMethod<Action<IAttackableUnit, double>>(Model, "Passive", "OnUpdate");
             onUpdate?.Invoke(this, diff);
+            foreach(var b in BuffGameScriptControllers)
+            {
+                b.Update(diff);
+                if (b.NeedsRemoved())
```

Comment:
It seems we need to consider doing dependency inversion here to avoid the infinite `NeedsRemoved` (TODO: change this method name) checks. (It's like busy waiting)

Reply:
I don't see the problem here. There is a finished number of elements in BuffGameScriptControllers

Reply:
Let's say I have buff that will be active for 10,000 ticks. Everytime you update and check for removal (aka busy-waiting). Instead, we can callback from the buff to avoid this busy-waiting.

Reply:
Open issue about this.